### ISSUE TITLE
EXC_BAD_INSTRUCTION INVOP in -[AWSTask runContinuations]

### ISSUE THREAD

**Issue by harlanhaskins** [aws/aws-sdk-ios#244]
Using the Xcode 7 GM, I'm getting an EXC_BAD_INSTRUCTION every time I complete a file upload:

```rust
let request = AWSS3TransferManagerUploadRequest()
request.body = NSURL.fileURLWithPath(tempFilePath)
request.key = filename
request.bucket = "bryx"
request.contentType = "image/jpeg"
request.ACL = .PublicRead
request.uploadProgress = { bytesSent, totalBytesSent, totalBytesExpectedToSend in
    progress?(Double(totalBytesSent) / Double(totalBytesExpectedToSend))
}
AWSS3TransferManager
    .defaultS3TransferManager()
    .upload(request)
    .continueWithBlock { task in
        completion(task?.error)
        return nil
    }
```

AWSTask.m, line 290
`[self.callbacks removeAllObjects]`
![screen shot 2015-09-11 at 6 30 17 pm](https://cloud.githubusercontent.com/assets/853032/9827718/29f9b75a-58b3-11e5-8a84-eeb23535fc82.png)

Debug build, no optimizations.

**Comment by harlanhaskins** [comment]
Exactly this code worked fine using Xcode 6.1.

**Comment by yosuke-matsuda** [comment]
I can run the code snippet on Xcode 7 GM without any crashes using AWS Mobile SDK for iOS 2.2.5. Since I do not have the implementations for `progress` and `completion`, I commented out these two lines. Have you created a test project with only the provided code snippet (and some setup code) in it and run it? Note that Xcode 7 has Swift 2, and some syntaxes have changes.

**Comment by harlanhaskins** [comment]
I think this is related to this: https://github.com/duemunk/Async/issues/46

**Comment by yosuke-matsuda** [comment]
I assume this has been resolved. If you continue to counter the issue, please feel free to update the thread with the requested information. Thanks.

**Comment by harlanhaskins** [comment]
Yes; apologies for not closing this. This was resolved in the Swift 2.1 update -- it had nothing to do with AWS. Thanks!
